File: 알고리즘/2/Intersection of Two Arrays 2.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> map;
        if(nums1.size() < nums2.size()){
            swap(nums1, nums2);
        }

        for(auto ele : nums1){
            map[ele]+=1;
        }
        vector<int> result;
        for (auto ele: nums2){
            if(map[ele]>0){
                map[ele]-=1;
                result.push_back(ele);
            }
        }
        sort(result.begin(), result.end());
        return result;
    }
};

File: 알고리즘/2/Intersection of Two Arrays 2.py

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:

        if len(nums1) < len(nums2):
            nums1, nums2 = nums2, nums1
        dict1 = {}
        for ele in nums1:
            dict1[ele] = dict1.get(ele, 0) + 1
        result = []
        for ele in nums2:
            if(dict1.get(ele, 0)!=0):
                dict1[ele]-=1
                result.append(ele)
        
            
        return sorted(result)


